### PR TITLE
[AKS] Fixed case sensitive issue for AKS addon and addon config

### PR DESCRIPTION
**Pull request by KaiqiZhang**
**Description**  

'az aks' has some case sensitive issues for addon profile names and addon profile configs. Two addon profiles for the same actual addon will be sent in the request if case is different. We have received customer report that kube-dashboard addon cannot be disable, due to this bug.
This PR is to make sure all the keys are lower cased before comparing, which is consistent with the AKS server side.

**Testing Guide**  

az aks disable-addons --addons kube-dashboard -g eastus -n testCluster

**History Notes**  

[AKS] Fixed case sensitive issue for AKS addon and addon config

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by yungezz** [comment]
hi @arrownj could you pls help to review? thanks

**Review by arrownj — COMMENTED** [review]
The code looks good to me. @KaiqiZhang , could you please help fix the CI failure first ? Thanks

**Comment by KaiqiZhang** [comment]
@arrownj Looks like the CI failure was not due to my change. Somehow rerun and passed.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -2155,11 +2158,12 @@ def aks_get_credentials(cmd, client, resource_group_name, name, admin=False,
         raise CLIError("Fail to find kubeconfig file.")
 
 
+# To be consistent, values of ADDONS dict should all be lower case
 ADDONS = {
```
Comment (robbiezhang) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2162]: can you add a simple unit test to ensure the values of this dictionary are lowercase?

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -1609,8 +1609,8 @@ def aks_browse(cmd, client, resource_group_name, name, disable_browser=False,
 
     # verify the kube-dashboard addon was not disabled
     instance = client.get(resource_group_name, name)
-    addon_profiles = instance.addon_profiles or {}
-    addon_profile = addon_profiles.get("kubeDashboard", ManagedClusterAddonProfile(enabled=True))
+    addon_profiles = {k.lower(): v for k, v in (instance.addon_profiles or {}).items()}
+    addon_profile = addon_profiles.get("kubedashboard", ManagedClusterAddonProfile(enabled=True))
```
Comment (robbiezhang) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:1613]: shall we define it as a const? actually, azure-cli-extensions has extract the consts into _consts file. consider to adopt the same.
https://github.com/Azure/azure-cli-extensions/blob/master/src/aks-preview/azext_aks_preview/_consts.py

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (KaiqiZhang) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:2162]: sure

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 2 above)
Comment (KaiqiZhang) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py:1613]: sure

**Review by arrownj — APPROVED** [review]

**Comment by arrownj** [comment]
The code is good to me. Please ping me to merge it if it's ready to merge from your side. @KaiqiZhang @robbiezhang

**Comment by KaiqiZhang** [comment]
Hold until we have a discussion about the naming convention.

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 3]
```diff
@@ -2155,11 +2158,12 @@ def aks_get_credentials(cmd, client, resource_group_name, name, admin=False,
         raise CLIError("Fail to find kubeconfig file.")
 
 
+# To be consistent, values of ADDONS dict should all be lower case
 ADDONS = {
-    'http_application_routing': 'httpApplicationRouting',
+    'http_application_routing': 'httpapplicationrouting',
```
Comment (ruchikaguptaa) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:2163]: keep the other addons as is so that it doesn't break customer scripts.

**Comment by KaiqiZhang** [comment]
@arrownj Sorry, after discussion we decided to keep the keys as it is. I have created another PR to quick fix the dashboard issue since it's the only thing broken right now. Please move to [https://github.com/Azure/azure-cli/pull/15123](https://github.com/Azure/azure-cli/pull/15123)

**Pull request closed without merging**